File: app/src/main/java/com/parabola/newtone/mvp/view/ArtistTracksView.java
package com.parabola.newtone.mvp.view;

import com.parabola.domain.model.Track;
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView;

import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.alias.AddToEndSingle;

@AddToEndSingle
public interface ArtistTracksView extends MvpView {
    void refreshTracks(List<Track> tracks);
    void setItemViewSettings(TrackItemView viewSettings);
    void setItemDividerShowing(boolean showed);
    void setSectionShowing(boolean enable);

    void setArtistName(String artistName);
    void setTracksCountTxt(String tracksCountStr);

    void setCurrentTrack(int trackId);

}

File: app/src/main/java/com/parabola/newtone/mvp/view/ArtistTracksView.kt
package com.parabola.newtone.mvp.view

import com.parabola.domain.model.Track
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle

@AddToEndSingle
interface ArtistTracksView : MvpView {
    fun refreshTracks(tracks: List<Track>)
    fun setItemViewSettings(viewSettings: TrackItemView)
    fun setItemDividerShowing(showed: Boolean)
    fun setSectionShowing(enable: Boolean)

    fun setArtistName(artistName: String)
    fun setTracksCountTxt(tracksCountStr: String)

    fun setCurrentTrack(trackId: Int)
}
